### PR TITLE
Enh/dimensionality tests

### PR DESCRIPTION
**Pull request by PatrickSampaioUSP**
## Pull request type

Please check the type of change your PR introduces:

- [x] Code base additions (bugfix, features)
- [ ] Code maintenance (refactoring, formatting, renaming, tests)
- [ ] ReadMe, Docs and GitHub maintenance
- [ ] Other (please describe):

## Pull request checklist

Please check if your PR fulfills the following requirements, depending on the type of PR:

- ReadMe, Docs and GitHub maintenance:

  - [ ] Spelling has been verified
  - [ ] Code docs are working correctly 

- Code base maintenance (refactoring, formatting, renaming):

  - [ ] Docs have been reviewed and added / updated if needed
  - [ ] Lint (`black rocketpy`) has passed locally and any fixes were made
  - [ ] All tests (`pytest --runslow`) have passed locally

- Code base additions (for bug fixes / features):

  - [x] Tests for the changes have been added
  - [ ] Docs have been reviewed and added / updated if needed
  - [ ] Lint (`black rocketpy`) has passed locally and any fixes were made
  - [ ] All tests (`pytest --runslow`) have passed locally

## What is the new behavior?

Added dimension tests for the Rocket class

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -47,6 +48,50 @@ def rocket(solid_motor):
...
 
 
+@pytest.fixture
+def m():
+    return numericalunits.m
+
+
+@pytest.fixture
+def kg():
+    return numericalunits.kg
```
Comment (giovaniceotto) [inline tests/conftest.py:58]: Should there by a fixture for seconds as well?

```suggestion
@pytest.fixture
def m():
    return numericalunits.m


@pytest.fixture
def kg():
    return numericalunits.kg


@pytest.fixture
def s():
    return numericalunits.s
```

File: tests/conftest.py [hunk 2]
```diff
@@ -47,6 +48,50 @@ def rocket(solid_motor):
...
+
+
+@pytest.fixture
+def dimensioneless_solid_motor(kg, m):
+    example_motor = SolidMotor(
+        thrustSource="data/motors/Cesaroni_M1670.eng",
+        burnOut=3.9,
```
Comment (giovaniceotto) [inline tests/conftest.py:65]: Should the burnOut time be given in seconds?

```suggestion
def dimensioneless_solid_motor(kg, m, s):
    example_motor = SolidMotor(
        thrustSource="data/motors/Cesaroni_M1670.eng",
        burnOut=3.9*s,
```

**Review by giovaniceotto — COMMENTED** [review]
Had a quick look and noticed no time dimensions are checked. Did we forget it or is this the intended behavior?

I suspect wonder if there are any calculations which involve a change in time dimensions (such as time derivatives or integration) in the Rocket class.

**Review by giovaniceotto — COMMENTED** [review]
From what I understand, when an equation does contain a dimensional error, an assert will fail. However, the assert can fail for multiple reasons, and a dimensional error is only one of the possibilities.

I wonder if we should implement such tests in a way that we have exclusive tests to guarantee dimensional homogeneity. This means that some tests would check only dimensionality and would fail if, and only if, there is a dimensional error. Another test would be needed verify if the function is producing a correct numerical answer.

The advantage of doing so is that the developer would have more information to understand where the error is and what it is.

File: tests/conftest.py (hunk 1 above)
Comment (Gui-FernandesBR) [reply tests/conftest.py:58]: Sure thing! But I think we use input based on time only on addParachutes method (the lag input) so in that case I think I agree using only mass and length for now

**Review by Gui-FernandesBR — APPROVED** [review]
I'm approving since it's OK for me, however Giovani's comments should be addressed before merging.
Great work btw, I think it's the beggining of a second level unit tests, PR&O is becoming expert on such things

**Comment by PatrickSampaioUSP** [comment]
@Gui-FernandesBR I think we should wait a little, I'm working on something regarding that comment

**Comment by PatrickSampaioUSP** [comment]
@giovaniceotto Could you take a look again, please?

**Review by giovaniceotto — APPROVED** [review]

**Pull request merged**